h/tasks/indexer.py

```diff
@@ -47,13 +47,15 @@ def sync_annotations(limit):
 
 
 @celery.task(acks_late=False)
-def report_sync_annotations_queue_length():
-    search_index = celery.request.find_service(name="search_index")
-
-    count = search_index._queue.count(
-        ["storage.create_annotation", "storage.update_annotation"]
-    )
-
-    newrelic.agent.record_custom_metric(
-        "Custom/SyncAnnotations/Queue/API/Length", count
+def report_job_queue_metrics():
+    queue = celery.request.find_service(name="search_index")._queue
+
+    newrelic.agent.record_custom_metrics(
+        [
+            (
+                "Custom/SyncAnnotations/Queue/API/Length",
+                queue.count(["storage.create_annotation", "storage.update_annotation"]),
+            ),
+            ("Custom/SyncAnnotations/Queue/Expired/Length", queue.count(expired=True)),
+        ]
```

Comment:
I'll have to rename this method in h-periodic now that I've renamed it here.

I think this should work, apparently `record_custom_metrics()` accepts an iterable of `(name, value)` 2-tuples: https://docs.newrelic.com/docs/agents/python-agent/python-agent-api/recordcustommetrics-python-agent-api

The New Relic metric names should have said "count" instead of "length" but I've already created a dashboard with several charts so I don't want to change that now

Reply:
Yeah this looks fine to me.